File: DSA/LakshaKundra_590013237/Day_65_Question_3.py
def min_removals(intervals):
    intervals.sort(key=lambda x: x[1])
    count, last_end = 0, float('-inf')
    for s, e in intervals:
        if s >= last_end:
            count += 1
            last_end = e
    return len(intervals) - count, count

print(min_removals([(1, 2), (2, 3), (3, 4), (1, 3)]))
print(min_removals([(1, 2), (1, 2), (1, 2)]))

File: DSA/VenikaPanwar_590013755/Day_65_Question_3.java
import java.util.*;

public class TaskScheduling {
    public static int removeMinTasks(int[][] intervals) {
        Arrays.sort(intervals, Comparator.comparingInt(a -> a[1]));
        int count = 1;
        int lastEnd = intervals[0][1];

        for (int i = 1; i < intervals.length; i++) {
            if (intervals[i][0] >= lastEnd) {
                count++;
                lastEnd = intervals[i][1];
            }
        }
        return intervals.length - count; // tasks removed
    }

    public static void main(String[] args) {
        int[][] tasks1 = {{1, 2}, {2, 3}, {3, 4}, {1, 3}};
        System.out.println("Remove " + removeMinTasks(tasks1)); // 1

        int[][] tasks2 = {{1, 2}, {1, 2}, {1, 2}};
        System.out.println("Remove " + removeMinTasks(tasks2)); // 2
    }
}
